Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+                    fActiveLines.push_back(new_Line);
+                    match = true;
+                }
+            }
+         }
+         return true;
+     }
+
+    bool KTSeqTrackFinder::EmitPreCandidate(LineRef Line)
+    {
+        bool LineIsTrack = true;
+
+        if (fApplyPowerCut)
+        {
+            if (Line.fAmplitudeSum <= fPowerThreshold)
+            {
+                LineIsTrack = false;
+            }
+        }
+        if (fApplyDensityCut)
+        {
+            if ((double)Line.fNPoints/(Line.fEndTimeInRunC-Line.fStartTimeInRunC)*1000.0 <= fDensityThreshold)
+            {
+                LineIsTrack = false;
+            }
+        }
+
+        if (LineIsTrack == true)
+        {
+            //KTDEBUG(stflog, "Line slope is "<<Line.fSlope);
+            //Line.FinishTrack();
+            //KTDEBUG(stflog, "Track slope is "<<Line.fSlope);
+
+
+            // Set up new data object
+            Nymph::KTDataPtr data( new Nymph::KTData() );
+            KTProcessedTrackData& newTrack = data->Of< KTProcessedTrackData >();
+            newTrack.SetComponent( Line.fComponent );
+            newTrack.SetTrackID(fNLines);
+            fNLines++;
+
+            newTrack.SetStartTimeInRunC( Line.fStartTimeInRunC );
+            newTrack.SetEndTimeInRunC( Line.fEndTimeInRunC );
+            newTrack.SetStartTimeInAcq( Line.fStartTimeInAcq);
+            newTrack.SetStartFrequency( Line.fStartFrequency );
+            newTrack.SetEndFrequency( Line.fEndFrequency );
+            newTrack.SetSlope(Line.fSlope);
+            newTrack.SetSlopeSigma(Line.fSlopeSigma);
+            newTrack.SetInterceptSigma(Line.fInterceptSigma);
+            newTrack.SetStartFrequencySigma(Line.fStartFrequencySigma);
+            newTrack.SetEndFrequencySigma(Line.fEndFrequencySigma);
+            newTrack.SetTotalPower(Line.fAmplitudeSum);
+
+            // Process & emit new track
+
+            KTINFO(stflog, "Now processing track candidate");
+            ProcessNewTrack( newTrack );
+
+            KTDEBUG(stflog, "Emitting track signal");
+            fTrackSignal( data );
+        }
+        else
+        {
+            KTDEBUG(stflog, "Line did not make it above the cut and was not emitted as track");
+        }
+        return true;
+    }
+
+    void KTSeqTrackFinder::ProcessNewTrack( KTProcessedTrackData& myNewTrack )
+    {
+        myNewTrack.SetTimeLength( myNewTrack.GetEndTimeInRunC() - myNewTrack.GetStartTimeInRunC() );
+        myNewTrack.SetFrequencyWidth( myNewTrack.GetEndFrequency() - myNewTrack.GetStartFrequency() );
+
+        // the slope is already calculated
+        //myNewTrack.SetSlope( myNewTrack.GetFrequencyWidth() / myNewTrack.GetTimeLength() );
+        myNewTrack.SetIntercept( myNewTrack.GetStartFrequency() - myNewTrack.GetSlope() * myNewTrack.GetStartTimeInRunC() );
+
+    }
+
+    void KTSeqTrackFinder::SearchTrueLinePoint(Point& Point, std::vector<double>& slice)
```

Comment:
1. Variable names should start with lower-case letters and use camelCase.  These need fixing: `Point`, `Delta`, `loop_counter`, `max_iterations`, `old_frequencybin` (also needs to be initialized when declared)

1. Use the prefix `++` to advance `loop_counter`